api/trace/api.go

```diff
@@ -189,6 +174,26 @@ func (sk SpanKind) String() string {
 	}
 }
 
+// WithParent explicitly sets the parent context. When the parent
+// context is not set explicitly, it uses a context passed to the
+// Start function.
+//
+// Figuring the parent span context from context should follow these
+// steps:
+//  1. If a remote context is set
+//     a. If a current span is set with the same trace_id, use the span's
+//        SpanContext as parent.
+//     b. If the current span belongs to a different trace_id, use the remote
+//        context as parent.
```

Comment:
Is condition 1.b tested? 
I looked at the tests but didn't find one. Maybe it is there and I just missed it.

Reply:
I had also proposed that the current trace_id be used to decide whether to take the current span or the remote context, and in that discussion was convinced (by @mwear) that we should do the simpler thing. Always use the span as parent if there is one, then fall back to the remote. It means that, in some cases, we'll have to clear the current span in order to use the remote context, which I think is acceptable.  See here: https://github.com/open-telemetry/opentelemetry-specification/pull/423